clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -77,19 +78,26 @@
     private boolean retry;
     private boolean reopened;
 
-    public ProducerBatch(TopicPartition tp, MemoryRecordsBuilder recordsBuilder, long createdMs) {
+    public ProducerBatch(final TopicPartition tp, final MemoryRecordsBuilder recordsBuilder, final long createdMs) {
         this(tp, recordsBuilder, createdMs, false);
     }
 
-    public ProducerBatch(TopicPartition tp, MemoryRecordsBuilder recordsBuilder, long createdMs, boolean isSplitBatch) {
+    public ProducerBatch(final TopicPartition tp, final MemoryRecordsBuilder recordsBuilder,
+                         final long createdMs, final boolean isSplitBatch) {
+        this(tp, recordsBuilder, createdMs, isSplitBatch, new ProduceRequestResult(tp), new ArrayList<>());
+    }
+
+    ProducerBatch(final TopicPartition tp, final MemoryRecordsBuilder recordsBuilder, final long createdMs,
+                  final boolean isSplitBatch, final ProduceRequestResult produceFuture, final List<ProducerBatch> childrenProducerBatch) {
```

Comment:
Why is childrenProducerBatch part of this constructor, if the only user of it is the other constructor? getChildrenProducer batch is public so I don't think it needs to be this way for testing either?

Reply:
In that way we decouple the children `List` as a collaborator (the constructor ins package-private), otherwise, I need to mimic the behavior of:
`public Deque<ProducerBatch> split(int splitBatchSize)` in order to populate the list of batches (children) that will be created as a consequence of the split process.